### DISCUSSION TITLE
[Kavita] Support PDF Metadata

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2545] [Ideas]
### Idea Description

Posted by Zackary Lowery
· Dec 2021
· 19 Upvotes
https://feats.kavitareader.com/posts/26/support-pdf-metadata

Description

Kavita should support parsing metadata from PDF documents. I use Calibre to organize and manage my ebooks, and Kavita parses the same filesystem location for it's media. This causes several issues:

    Calibre limits file name length due to Windows' limited file system path length
    Calibre removes special characters from the file name that may cause issues on certain filesystems
    Additional metadata, such as summaries/descriptions/tags cannot reasonably fit in a file name

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by DieselTech** [comment]
lackafi
· Jul 2022

There is a C# project called PdfPig that may be relevant (link below). I'd love it if I could extract data from the calibre opf files, write them into the pdf metadata using something like pdftk and then have Kavita parse it and extract e.g. authors from it.

https://github.com/UglyToad/PdfPig

**Comment by Cducharme84** [comment]
Rather than do a separate FR as I had forgotten this one exists, as well as the very relevant but highly specific [idea](https://github.com/Kareadita/Kavita/discussions/3103) found here. 

There's been non-Calibre focused tooling that's become available that would be very useful for the PDF heavy servers that need to keep the format for various reasons usually due to flexibility with tabling and image vectoring, comicbox from AJ of Codex is currently able to write a comicinfo.xml into the PDF file's [keywords](https://github.com/Kareadita/Kavita/discussions/3103) allowing for a format of metadata Kavita is already equipped to handle to be parsed.

I know Joe's had a discussion with AJ briefly about this but I also know how easy it is for any individual piece of information to be forgotten when there's just so much. Since the PDF refactor is still impending, I did think it'd be a good idea to have a reminder for Joe of the additional tooling and their methodologies of storing and parsing PDF metadata for consideration during the planning stage.

[Comicbox main project:](https://github.com/ajslater/comicbox)

**Comment by majora2007** [comment]
This was delivered with v0.8.6.
